lib/ui/viewer/search/search_widget.dart

```diff
@@ -139,34 +143,63 @@ class _SearchWidgetState extends State<SearchWidget> {
     );
   }
 
-  Future<List<SearchResult>> getSearchResultsForQuery(String query) async {
-    final List<SearchResult> allResults = [];
+  @override
+  void dispose() {
+    if (_debounce != null) {
+      _debounce.cancel();
+    }
+    super.dispose();
+  }
 
-    final queryAsIntForYear = int.tryParse(query);
-    if (_isYearValid(queryAsIntForYear)) {
-      final yearResult =
-          await _searchService.getYearSearchResults(queryAsIntForYear);
-      allResults.add(yearResult); //only one year will be returned
+  Future getSearchResultsForQuery(String query) async {
+    final List<SearchResult> allResults = [];
+    if (query.isEmpty) {
+      if (_debounce != null && _debounce.isActive) {
+        _debounce.cancel();
+      }
+      return (allResults);
     }
 
-    final holidayResults = await _searchService.getHolidaySearchResults(query);
-    allResults.addAll(holidayResults);
+    final completer = Completer();
+
+    _debounceQuery(() async {
+      final queryAsIntForYear = int.tryParse(query);
```

Comment:
This could be extracted to a separate function?

Reply:
Tried this before making the PR and ran into some issues, will try again.